monai/networks/blocks/squeeze_and_excitation.py

```diff
@@ -87,6 +95,88 @@ def __init__(
     def forward(self, x: torch.Tensor):
         """
         Args:
-            x: in shape (batch, channel, spatial_1[, spatial_2, ...]).
+            x: in shape (batch, in_channels, spatial_1[, spatial_2, ...]).
         """
         return x + super().forward(x)
+
+
+class SEBlock(nn.Module):
+    """
+    Residual module enhanced with Squeeze-and-Excitation::
+
+        ----+- conv1 --  conv2 -- conv3 -- SE -o---
+            |                                  |
+            +---(channel project if needed)----+
+
+    Re-implementation of the SE-Resnet block based on:
+    "Hu et al., Squeeze-and-Excitation Networks, https://arxiv.org/abs/1709.01507".
+    """
+
+    def __init__(
+        self,
+        spatial_dims: int,
+        in_channels: int,
+        n_chns_1: int,
+        n_chns_2: int,
+        n_chns_3: int,
+        conv_param_1: Optional[dict] = None,
+        conv_param_2: Optional[dict] = None,
+        conv_param_3: Optional[dict] = None,
+        r: int = 2,
+        acti_type_1: str = "relu",
```

Comment:
Could you please list all the acceptable types in doc-string?

Reply:
agreed, the layer factory class (Act/Norm/Conv) docstrings may need some 'runtime' assignment, I'll create a separate issue/PR to address that @ericspod